### ISSUE TITLE
Word64.fromInt sign extension

### ISSUE THREAD

**Issue by vqns** [melsman/mlkit#181]
`Word64.fromInt` does not sign extend its argument if the default Int structure is Int63.

    $ cat t.sml
    val _ = print ((Word64.toString o Word64.fromInt) ~1 ^ "\n")
    $ mlkit --no_gc t.sml >/dev/null
    $ ./run
    FFFFFFFFFFFFFFFF
    $ mlkit t.sml >/dev/null
    $ ./run
    7FFFFFFFFFFFFFFF

**Comment by melsman** [comment]
Nice catch!! ;)

**Comment by melsman** [comment]
The definition of `fromInt` in `Word64` is wrong:

```
    fun cast_iw (a: int) : word = prim("id", a)
    fun fromInt (i : int) : word64 = prim("__word_to_word64", cast_iw i)
```

**Comment by melsman** [comment]
Maybe we can use the following definition:

```sml
    fun fromInt (x: int) : word64 = prim("__int_to_int64", x)
```

**Comment by melsman** [comment]
Fixed by #180
